bin/Infer.ml

```diff
@@ -76,7 +80,7 @@ let command =
         let states = List.(permute
           ~random_state:(Random.State.make [| 79 ; 97 |])
           (map (Stdio.In_channel.input_lines states_chan)
-              ~f:(fun l -> map (String.split ~on:'\t' l) ~f:Value.of_string)))
+              ~f:(fun l -> map (map ~f:(fun x -> (Sexp.of_string x)) (String.split ~on:'\t' l)) ~f:Value.of_string)))
```

Comment:
Why is this change needed?

Reply:
since Value.of_string takes an sexp. Maybe I can do the change I did for Type.of_sexp, in case of Value.of_string.

Reply:
Oh I see. Here is what I suggest:
1. Change that function name to `Value.of_sexp` (and similarly `Type.of_sexp`)
2. Add 2 new functions `Value.of_string` and `Type.of_string` which parse the string to an s-expression and then call the corresponding `of_sexp`
3. Revert this change